core-kotlin-modules/core-kotlin-datastructures/src/main/kotlin/com/baeldung/insertionsort/SortIntArray.kt

```diff
@@ -0,0 +1,23 @@
+fun insertionSort(arr: IntArray) {
+    val n = arr.size
+    for (i in 1 until n) {
+        val key = arr[i]
+        var j = i - 1
+        while (j >= 0 && arr[j] > key) {
+            arr[j + 1] = arr[j]
+            j--
+        }
+        arr[j + 1] = key
+    }
+}
+
+fun insertionSortWithIntArray() {
+    val myArray = intArrayOf(5, 2, 9, 3, 6)
+    insertionSort(myArray)
+    println("Sorted Array:")
+    for (element in myArray) {
+        println(element)
+    }
```

Comment:
Same with this part, it feels like we should have a test that is asserting the final order of the now sorted array.